x-pack/plugins/reporting/public/components/report_listing.tsx

```diff
@@ -0,0 +1,315 @@
+        name: 'Status',
+        render: (status: string, record: Job) => {
+          let maxSizeReached;
+          if (record.max_size_reached) {
+            maxSizeReached = <span> - max size reached</span>;
+          }
+          let statusTimestamp;
+          if (status === 'processing' && record.started_at) {
+            statusTimestamp = this.formatDate(record.started_at);
+          } else if (record.completed_at && (status === 'completed' || status === 'failed')) {
+            statusTimestamp = this.formatDate(record.completed_at);
+          }
+          return (
+            <div>
+              {status}
+              {' at '}
+              <span className="eui-textNoWrap">{statusTimestamp}</span>
+              {maxSizeReached}
+            </div>
+          );
+        },
+      },
+      {
+        name: 'Actions',
+        actions: [
+          {
+            render: (record: Job) => {
+              return (
+                <div className="euiTableCellContent__hoverItem">
+                  {this.renderDownloadButton(record)}
+                  {this.renderReportErrorButton(record)}
+                </div>
+              );
+            },
+          },
+        ],
+      },
+    ];
+
+    const pagination = {
+      pageIndex: this.state.page,
+      pageSize: 10,
+      totalItemCount: this.state.total,
+      hidePerPageOptions: true,
+    };
+
+    return (
+      <EuiBasicTable
+        itemId={'id'}
+        items={this.state.jobs}
+        loading={this.state.isLoading}
+        columns={tableColumns}
+        noItemsMessage={this.state.isLoading ? 'Loading reports' : 'No reports have been created'}
+        pagination={pagination}
+        onChange={this.onTableChange}
+      />
+    );
+  }
+
+  private renderDownloadButton = (record: any) => {
+    if (record.status !== 'completed') {
+      return;
+    }
+
+    const button = (
+      <EuiButtonIcon
+        onClick={() => downloadReport(record.id)}
+        iconType="importAction"
+        aria-label="Download report"
+      />
+    );
+
+    if (record.max_size_reached) {
+      return (
+        <EuiToolTip position="top" content="Max size reached, contains partial data.">
+          {button}
+        </EuiToolTip>
+      );
+    }
+
+    return button;
+  };
+
+  private renderReportErrorButton = (record: any) => {
+    if (record.status !== 'failed') {
+      return;
+    }
+
+    return <ReportErrorButton jobId={record.id} />;
+  };
+
+  private onTableChange = ({ page }: { page: any }) => {
+    const { index: pageIndex } = page;
+
+    this.setState(
+      {
+        page: pageIndex,
+      },
+      this.fetchJobs
+    );
+  };
+
+  private fetchJobs = async () => {
+    // avoid page flicker when poller is updating table - only display loading screen on first load
+    if (this.isInitialJobsFetch) {
+      this.setState({ isLoading: true });
+    }
+
+    let jobs;
+    let total;
+    try {
+      jobs = await jobQueueClient.list(this.state.page);
```

Comment:
Can you type jobs and total so they aren't `any` and add them to the return values of `jobQueueClient`?

Reply:
The return is just the es document type. Not sure this needs to be typed out.

Reply:
Ideally kfetch would take in a type variable which users would then pass in as the expected return shape (see https://www.typescriptlang.org/docs/handbook/generics.html).  But since that would take more effort to migrate over, you can have the jobQueueClient mark it's return values, like:

`public list = (page = 0, jobIds?: string[]): Array<JobResponse> => {`

Any time you do something like:

`let aVariableName` without explicitly assigning the type, via
`let aVariableName: string;` it's any.  e.g. total is `any` here, and that should be a number.  I think even if you type the returns of the jobQueueClient you'll have to type these variables to avoid `any` being assigned.